tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -116,14 +116,16 @@ def model_to_dot(model, show_shapes=False, show_layer_names=True, rankdir='TB'):
 
     # Rebuild the label as a table including input/output shapes.
     if show_shapes:
+      def format_shape(shape):
+        return str(shape).replace(str(None), '?')
```

Comment:
Thanks for wanting to improve the rendering!

However, it would not match TensorFlow conventions, or Python conventions, to replace `None` with `?`.

Reply:
Yeah, but this is for rendering a pictorial representation of the model and usually not for machine consumption, so aesthetic should be more important?

Reply:
@dynamicwebpaige  By the way, I just tried

```python
    for name, tensor in zip(model.input_names, model.inputs):
        print(name, ':', tensor.dtype.name, tensor.shape)
```
which produces output:
```
decoder_inputs : int32 (?, 19)
```

Therefore, the Keras/TF `tensor.shape` is already formatted to use `?` instead of `None`

Reply:
@rthadur may you request another review on this? thanks